### PR TITLE
Bug fix - comparing identical float expressions on linux32

### PR DESCRIPTION
**Pull request by e-kayrakli**
Solves the bug reported by @daviditen - #5598

Specifically:
- Prevents removing temps that are used in float comparisons - this solves #5598 
- Changes the signature of helper function `isIntegerPromotionPrimitive`
- Fixes typo in a comment
- Retires the associated future.

Locally tested test/release/examples/hello* with std linux64 and linux32

### CODE REVIEW

**Comment by daviditen** [comment]
@e-kayrakli 

Yes, please do remove the .future, .bad, and .skipif files for the fixed .future test.

Let me know when you feel this is ready to be reviewed and merged and I'll run a full Linux64 test run for you.

**Comment by e-kayrakli** [comment]
Thanks @daviditen - I removed the files (that was all that's necessary to make it a test, right?) and made sure that the test passes on linux32. I think it is ready for a full test.

**Comment by e-kayrakli** [comment]
..also updated the PR message to reflect the change

**Comment by daviditen** [comment]
Yes, removing those 3 files turns it into a normal test.  I'll get a test run going in a bit.

Thanks!

**Comment by daviditen** [comment]
Sorry this took a bit longer than I expected.  I ran a full Linux64 test and it came back clean.

File: compiler/codegen/expr.cpp [hunk 1]
```diff
@@ -4416,7 +4416,8 @@ GenRet CallExpr::codegenPrimitive() {
       // be enough to cast integers that are smaller than standard C int for
       // target architecture. However, there was no easy way of obtaining that
       // at the time of writing this piece. Engin
-      if (dst == src && !(is_int_type(dst) || is_uint_type(dst)) ) {
+      if (dst == src && !(is_int_type(dst) || is_uint_type(dst) ||
+            is_real_type(dst)) ) {
```
Comment (daviditen) [inline compiler/codegen/expr.cpp]: Could you please move is_real_type over so it starts in the same column as is_int_type?

File: compiler/passes/denormalize.cpp [hunk 2]
```diff
@@ -338,7 +339,8 @@ bool isDenormalizable(Symbol* sym,
                   (ce->isPrimitive(PRIM_ARRAY_SHIFT_BASE_POINTER) && ce->get(1) == se) ||
                   (ce->isPrimitive(PRIM_MOVE) &&
                    ce->get(1)->typeInfo() !=
-                   ce->get(2)->typeInfo()))) {
+                   ce->get(2)->typeInfo()) ||
+                  isFloatComparisonPrimitive(ce))) {
```
Comment (daviditen) [inline compiler/passes/denormalize.cpp]: ... and move isFloatComparisonPrimitive 1 column to the right.

**Comment by e-kayrakli** [comment]
Nice!

I just made the changes you requested.

**Comment by daviditen** [comment]
Thanks!  IMO this is ready to merge then.  I can merge it for you early next week if you agree.

**Comment by e-kayrakli** [comment]
Sure, no problem!

**Pull request merged**